### ISSUE TITLE
【Question】当输入为sequnce类型时classification_cost的weight参数

### ISSUE THREAD

**Issue by pkuyym** [PaddlePaddle/Paddle#1735]
模型为seqToseq模型，但是数据的格式为
src_seq  trg_seq  weight
希望使用weighted cost，在使用classification_cost时，指定weight类型为dense_vector(1)，会报异常
比如设置batch_size为5000，异常为：
Check failed: height_ == c.height_ (39792 vs. 5000)

**Comment by helinwang** [comment]
@pkuyym 从log看貌似您的输入batch size是39792而不是5000? 能不能把您的网络和reader实现贴上来看一下？

**Comment by pkuyym** [comment]
@helinwang 谢谢，我贴一下主要逻辑，我怀疑是不是序列数据的问题。
reader：
def process(settings, file_name):
    with open(file_name, 'r') as f:
        for line_count, line in enumerate(f):
            line_split = line.strip().split('\t')
            if settings.job_mode and len(line_split) != 3 and len(line_split) != 2:
                continue
            src_seq = line_split[0]  # one source sequence
            src_ids = _get_ids(src_seq, settings.src_dict)
            if settings.job_mode:
                trg_seq = line_split[1]  # one target sequence
                uni_trg_seq = unicode(trg_seq, 'gbk', errors = 'ignore')
                trg_chars = []
                for i in xrange(len(uni_trg_seq)):
                    trg_chars.append(uni_trg_seq[i])
                trg_ids = [settings.trg_dict.get(c.encode('gbk', errors = 'ignore'), UNK_IDX) for c in trg_chars]
                # remove sequence whose length > 100 in training mode
                if len(src_ids) > 100 or len(trg_ids) > 100:
                    continue
                trg_ids_next = trg_ids + [settings.trg_dict[END]]
                trg_ids = [settings.trg_dict[START]] + trg_ids
                if len(line_split) == 3:
                    weight = float(line_split[2]) / settings.scale_factor
                else:
                    weight = 1.0
                yield {
                    'source_language_word': src_ids,
                    'target_language_word': trg_ids,
                    'target_language_next_word': trg_ids_next,
                    'seq_weight' : [weight]
                }
            else:
                yield {'source_language_word': src_ids, 'sent_id': [line_count]}

网络优化参数：
settings(
    learning_method = AdamOptimizer(),
    batch_size = 5000,
    learning_rate = 5e-3,
    gradient_clipping_threshold = 25)

网络cost部分：
lbl = data_layer(name = 'target_language_next_word', size = target_dict_dim)
seq_weight = data_layer(name = 'seq_weight', size = 1)
cost = classification_cost(input = decoder, label = lbl, weight = seq_weight)

现在报错信息成为了：
F0406 15:14:32.385094 27170 BaseMatrix.cu:123] Check failed: height_ == c.height_ (39629 vs. 5000)
猜测是跟序列长度有关系

**Comment by pkuyym** [comment]
@helinwang hi，请问有解决方案吗

**Comment by guoshengCS** [comment]
Close since no update. Please reopen it if necessary.

**Comment by helinwang** [comment]
@guoshengCS Sorry it's my bad that I have not follow up with @pkuyym . The problem is not actually solved. I will reopen this issue.
